### DISCUSSION TITLE
Problems with data loading in q-form and validation in q-input

### DISCUSSION THREAD

**Discussion by thonhaus** [quasarframework/quasar#12444] [General - Components / Directives / etc] [answered]
Hello,

I've problems with data loading and the validation in q-input fields.

I try to load data with an axios call to the backend in the created() method of the component. It's then setted into the model, but the validation always fails. It seems that it doesn't recognize that there is already a value in the field.

Example:
https://codepen.io/thonhaus/pen/YzEQqRV

What's wrong?

Yours
Thomas

**Comment by MilosPaunovic** [comment]
Use `lazy-rules` prop on your `QInput`
[https://codepen.io/milospaunovic/pen/yLPXJOL?editors=1111](https://codepen.io/milospaunovic/pen/yLPXJOL?editors=1111)

**Comment by thonhaus** [comment]
`lazy-rules` doesn't solve the problem. If you focus the field and leave again without any changes, you'll see the validation will fail.

**Reply by MilosPaunovic** [reply]
If the value you've provided does not pass the rule, validation will fail.

**Reply by thonhaus** [reply]
The validation is "val && val.length > 0" and the inserted value is "100", which should pass the validation.

If you enter the field and change the value to e. g. "10", it will also pass.

**Comment by MilosPaunovic** [comment, marked as answer]
In your mounted hook you change the value like this:
`this.c_childs = 100`
and the rule is this:
`[ val => val && val.length > 0 || 'Bitte einen Wert eingeben!']`

Meaning you're trying to check length property of number, which returns `undefined`, so the error is trigerred.
If you want the value to be number, you're gonna have to re-write the rules to this:
`[ val => val && val > 0 || 'Bitte einen Wert eingeben!']`

**Reply by thonhaus** [reply]
Sorry, you're absolutely right. Too easy.... Sorry!
